Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabFocusInterface.h

```diff
@@ -47,6 +47,11 @@ namespace AzToolsFramework::Prefab
         //! @return The focused instance.
         virtual InstanceOptionalReference GetFocusedPrefabInstance(AzFramework::EntityContextId entityContextId) const = 0;
 
+        //! Returns whether the currently focused prefab instance is read-only.
+        //! @param EntityContextId The entity context id.
+        //! @return True if the currently focused prefab instance is read-only, false otherwise.
+        virtual bool IsFocusedPrefabInstanceReadOnly(AzFramework::EntityContextId entityContextId) const = 0;
```

Comment:
I thought proc. prefabs can't be focused upon since they shouldn't be modifying the prefab template. If they can't be focused upon, then we wouldn't run into the issue. Did any focus behavior change recently or were proc prefabs always allowed to be focused upon?

Reply:
I can see the value in inspecting proc prefab contents but maybe instead of focusing on it, we can use the expand icons to unfold the prefab and look at its content similar to what the override prototype does? 

In my head, it sounds like a cleaner way to deal with this problem because you can never edit the template if you never focus upon it. I agree that blocking these on the backend also should be done.

Reply:
Yes that was the long term approach, but this was implemented last year when overrides was still ways away.
I can see the use in both honestly - expand to override, inspect to see the original values.